drivers/video/fbdev/exynos/panel/mdnie.c

```diff
@@ -492,10 +492,10 @@ static void mdnie_update_scr_white_mode(struct mdnie_info *mdnie)
 	if (mdnie_mode == MDNIE_SCENARIO_MODE) {
 		if ((IS_LDU_MODE(mdnie)) && (mdnie->props.scenario != EBOOK_MODE)) {
 			mdnie->props.scr_white_mode = SCR_WHITE_MODE_ADJUST_LDU;
-		} else if (mdnie->props.update_sensorRGB ||
-			   (mdnie->props.mode == AUTO &&
-			    (mdnie->props.scenario == BROWSER_MODE ||
-			     mdnie->props.scenario == EBOOK_MODE))) {
+		} else if (mdnie->props.update_sensorRGB &&
+				mdnie->props.mode == AUTO &&
+				(mdnie->props.scenario == BROWSER_MODE ||
+				 mdnie->props.scenario == EBOOK_MODE)) {
```

Comment:
It appears that this patch causes troubles with dark mode on samsung firmwares.

Reply:
You can make a los config.

Reply:
I doubt that los config is a good idea. I will not test the kernel on los. Every los has it's own kernel which is a best choice for that rom.

Reply:
If you look at Ivan's kernel, it is stock with some patches (majority are here)

Reply:
Ok, this a revert for https://github.com/CruelKernel/samsung-exynos9820/commit/976682257c7deddec6ab02992bec7fe126ad22e5 Yes, users reported me that it causes problems on samsung firmware. I will just drop it from devel branch instead of reverting.

Reply:
Yes it does, on my s10 on stock too. And yes it's a revert. It's weird that on other devices like J5 it works with that commit, has android 9 with dark mode and oled screen